mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -920,6 +989,57 @@ LogicalResult tosa::MatMulOp::inferReturnTypeComponents(
   return success();
 }
 
+LogicalResult MatMulOp::verify() {
+  auto aType = llvm::dyn_cast<ShapedType>(getA().getType());
+  auto bType = llvm::dyn_cast<ShapedType>(getB().getType());
+
+  // Must be shaped tensor types
+  if (!aType) {
+    emitOpError("expect a shaped tensor for input a, got ") << getA().getType();
```

Comment:
`emitOpError` returns an `InFlightDiagnositc` which can be implicitly converted to a `LogicalResult`. In the case of a non-empty diagnostic this will evaluate to `failure()`, so no need to return failure() below, the result of the function call can be returned directly. This means we can also remove the `{}` on the if to align with the llvm style guide.

Reply:
done.

Reply:
Sorry what I meant was that you could write this as:
```

return mitOpError("expect a shaped tensor for input a, got ") << getA().getType();
```

Reply:
ah, thanks. Updated the code and other similar cases.